src/Servers/Kestrel/Core/src/Internal/ConfigurationReader.cs

```diff
@@ -388,37 +389,46 @@ internal CertificateConfig()
     public IConfigurationSection? ConfigSection { get; }
 
     // File
+
+    [MemberNotNullWhen(true, nameof(Path))]
     public bool IsFileCert => !string.IsNullOrEmpty(Path);
 
-    public string? Path { get; set; }
+    public string? Path { get; init; }
+
+    public string? KeyPath { get; init; }
 
-    public string? KeyPath { get; set; }
+    public string? Password { get; init; }
 
-    public string? Password { get; set; }
+    /// <remarks>
+    /// Vacuously false if this isn't a file cert.
```

Comment:
```suggestion
    /// Do not read this from config. This is set internally only for change detection.
    /// Vacuously false if this isn't a file cert.
```

Reply:
What does "from config" mean in this context?

Reply:
```Used for change tracking - not actually part of configuring the certificate.```